packages/SystemUI/res/values/aosip_config.xml

```diff
@@ -43,4 +43,8 @@
     <bool name="config_statusBarBurnInProtection">true</bool>
     <integer name="config_shift_interval">40</integer>
 
+    <!-- Whether the binned brightness sensor reports a value of 0 on proximity
+         near events -->
+    <bool name="doze_brightness_sensor_reports_proximity" translatable="false">true</bool>
```

Comment:
Doesn't most phones need to have this set to false?

Reply:
Dumpling for example never returns a 0

Reply:
Nah, the default behaviour is reporting zero values with `findSensorWithType(mSensorManager, mContext.getString(R.string.doze_brightness_sensor_type));`
And the phone that has crap proximity don't need that. so that's why the default is true

Reply:
https://github.com/DerpLab/platform_frameworks_base/pull/39/commits/31248562d7a6f0e9ffad95f0c75d6f7ab37b350a#diff-79c80ee601682fb3b305a531000217dcR178
https://github.com/DerpLab/platform_frameworks_base/pull/39/commits/31248562d7a6f0e9ffad95f0c75d6f7ab37b350a#diff-79c80ee601682fb3b305a531000217dcL300

Reply:
There is no "default" here as every device is diff. Also, this is related to blobs imo. It's a software thing to report 0 on proximity near events. When light is clearly never really 0

Reply:
Any way the question is what is the most common behavior to have as little new DT overlays as we can

Reply:
> [3124856#diff-79c80ee601682fb3b305a531000217dcR178](https://github.com/DerpLab/platform_frameworks_base/commit/31248562d7a6f0e9ffad95f0c75d6f7ab37b350a#diff-79c80ee601682fb3b305a531000217dcR178)
> [3124856#diff-79c80ee601682fb3b305a531000217dcL300](https://github.com/DerpLab/platform_frameworks_base/commit/31248562d7a6f0e9ffad95f0c75d6f7ab37b350a#diff-79c80ee601682fb3b305a531000217dcL300)

yea yea I read the code

Reply:
if proximity hardware is correct, the proximity will gated itself based on sensor type.
but in this elliptic case, such things is not exist and we can't bypass sensor finding from xml.

Reply:
> if proximity hardware is correct, the proximity will gated itself based on sensor type.
> but in this elliptic case, such things is not exist and we can't bypass sensor finding from xml.

the code _doesn't_ check for the physical h/w. It uses said xml value. Hence I'm asking a simple question. What is the behavior on most devices. My device for example. Won't turn in a 0. And hence needs to overlay this value to false in dt. If this is the case for most devices we should flip it so less devices need an overlay. Is all I'm saying.

Reply:
> Anyway the question is what is the most common behavior to have as little new DT overlays as we can

but why tho ? it's going back to default behavior if `doze_brightness_sensor_reports_proximity` is not setup, only the dev that know they phone will set it to false

Reply:
> There is no "default" here as every device is diff.

Yes it is, https://github.com/DerpLab/platform_frameworks_base/pull/39/commits/31248562d7a6f0e9ffad95f0c75d6f7ab37b350a#diff-79c80ee601682fb3b305a531000217dcR310

The default behavior is using `DozeSensors.findSensorWithType();` so it use a prox gated brightness sensor if available.

`doze_brightness_sensor_reports_proximity = false` is to bypass that.
if you don't want to bypass it, then just leave them alone.

Reply:
but it is. right here in the code I replied to above